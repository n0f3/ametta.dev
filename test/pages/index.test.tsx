import React from 'react';
import { render, fireEvent } from '../testUtils';
import Home from '../../pages/index';
import { RenderResult } from '@testing-library/react';

describe('Home Page', () => {
  it('matches snapshot', () => {
    const { asFragment }: RenderResult = render(<Home />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  // it('clicking button triggers alert', () => {
  //   const { getByText } = render(<Home />, {})
  //   window.alert = jest.fn()
  //   fireEvent.click(getByText('Test Button'))
  //   expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  // })
});
