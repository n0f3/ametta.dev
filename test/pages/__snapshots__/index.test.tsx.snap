// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <img
        alt="[Your Name]"
        class="headerHomeImage borderCircle"
        src="/images/profile.jpg"
      />
      <h1
        class="heading2Xl"
      >
        [Your Name]
      </h1>
    </header>
    <main>
      <section
        class="headingMd"
      >
        <p>
          [Your Self Introduction]
        </p>
        <p>
          (This is a sample website - youâ€™ll be building a site like this in 
          <a
            href="https://nextjs.org/learn"
          >
            our Next.js tutorial
          </a>
          .)
        </p>
      </section>
      <section
        class="headingMd padding1px"
      >
        <h2
          class="headingLg"
        >
          Blog
        </h2>
        <ul
          class="list"
        />
      </section>
    </main>
  </div>
</DocumentFragment>
`;
